core/src/main/java/apoc/cypher/Timeboxed.java

```diff
@@ -52,6 +57,12 @@ public Stream<MapResult> runTimeboxed(@Name("statement") String cypher, @Name("p
                 txAtomic.set(innerTx);
                 Result result = innerTx.execute(cypher, params == null ? Collections.EMPTY_MAP : params);
                 while (result.hasNext()) {
+                    if (Util.transactionIsTerminated(terminationGuard)) {
+                        txAtomic.get().close();
+                        offerToQueue(queue, POISON, timeout);
```

Comment:
I don't really understand the Posion part here, if the tx is terminated would we not want to just return the same warning as below? I might be missing some context here, so feel free to explain it to me :)

Reply:
A bit strange but without the poison queue, the transaction is marked as terminated but actually continues the execution until it finishes, and after that throws an exception `The transaction has been terminated...`

Anyway, I realized that the `TransactionTestUtil.checkTerminationGuard` could potentially produce a false positive, therefore I also added the execution time check (which in fact fails without this `POISON`)